### PR TITLE
Add ECS application examples

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

Add examples for ECS applications under some of the following strategies:
- Simple ( only `SYNC` stage )
- Canary
- BlueGreen

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: examples/ecs/bluegreen/taskdef.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      - containerPort: 80
+compatibilities:
+  - FARGATE
+requiresCompatibilities:
+  - FARGATE
+networkMode: awsvpc
+memory: 512
+cpu: 256
+pidMode: ""
+status: ACTIVE
```
Comment (nghialv) [inline examples/ecs/bluegreen/taskdef.yaml:20]: Is this `status` field can be specified by the user?

File: examples/ecs/canary/.pipe.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        containerPort: 80
+  pipeline:
+    stages:
+      # Rollout CANARY variant's workload.
+      # scale represents the percentage of workload will be rolled out,
+      # in this case 10% of the current PRIMARY variant's workload is set.
+      # Note: Rolled out variant serves no traffic at this time.
+      - name: ECS_CANARY_ROLLOUT
+        with:
+          scale: 10
```
Comment (nghialv) [inline examples/ecs/canary/.pipe.yaml:30]: 10% of workloads will receive 20% of traffic is not a best practice so I think we can change this to 30%.

File: examples/ecs/simple/.pipe.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+kind: ECSApp
+spec:
+  input:
+    serviceDefinitionFile: servicedef.yaml
+    taskDefinitionFile: taskdef.yaml
+    targetGroups:
+      primary:
+        targetGroupArn: arn:aws:elasticloadbalancing:ap-northeast-1:XXXX:targetgroup/ecs-lb/YYYY
+        containerName: web
+        containerPort: 80
```
Comment (nghialv) [inline examples/ecs/simple/.pipe.yaml:11]: Do we still need this targetGroups in case of using QuickSync?

Comment (khanhtc1202) [reply examples/ecs/simple/.pipe.yaml:11]: Yes, it is 🙏 When creating a task set, we have to point to the creating task set by a target group to enable those tasks to serve traffic. Without this target group setting, there is no way to access to the created task set 👀

File: examples/ecs/canary/.pipe.yaml (hunk 2 above)
Comment (khanhtc1202) [reply examples/ecs/canary/.pipe.yaml:30]: Nice catch ⚾

File: examples/ecs/bluegreen/taskdef.yaml (hunk 1 above)
Comment (khanhtc1202) [reply examples/ecs/bluegreen/taskdef.yaml:20]: Yes, it is 🙏 
```go
	// The status of the task definition.	
	Status TaskDefinitionStatus
```
ref: https://github.com/aws/aws-sdk-go-v2/blob/main/service/ecs/types/types.go#L3308-L3309

Comment (khanhtc1202) [reply examples/ecs/bluegreen/taskdef.yaml:20]: Oops, I get your point 👍 We're not going to use this anywhere else while registered this TaskDefinition 🙆‍♀️ Will remove it from this sample 👍

File: examples/ecs/canary/.pipe.yaml (hunk 2 above)
Comment (khanhtc1202) [reply examples/ecs/canary/.pipe.yaml]: addressed on https://github.com/pipe-cd/pipe/pull/2079/commits/fd360ed6ee4989ef6e41ce9e71630531b7c3ac8b 🙏

File: examples/ecs/bluegreen/taskdef.yaml (hunk 1 above)
Comment (khanhtc1202) [reply examples/ecs/bluegreen/taskdef.yaml]: addressed on https://github.com/pipe-cd/pipe/pull/2079/commits/fd360ed6ee4989ef6e41ce9e71630531b7c3ac8b 🙏

Comment (nghialv) [reply examples/ecs/bluegreen/taskdef.yaml]: 👍

**Comment by nghialv** [comment]
Btw, I think we can remove "piped" from the PR title.

**Comment by khanhtc1202** [comment]
> Btw, I think we can remove "piped" from the PR title.

No objection 🙆‍♀️

**Review by nakabonne — APPROVED** [review]
there you go

**Pull request merged**